LgiHdcpProfile/LgiHdcpProfile.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+* If not stated otherwise in this file or this component's LICENSE
+* file the following copyright and licenses apply:
+*
+* Copyright 2019 RDK Management
+* Copyright 2021 Liberty Global Service B.V.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+#include <string>
+
+#include "LgiHdcpProfile.h"
+
+#include "videoOutputPort.hpp"
+#include "videoOutputPortConfig.hpp"
+#include "dsMgr.h"
+#include "manager.hpp"
+#include "host.hpp"
+
+#include "UtilsJsonRpc.h"
+
+#define HDCP_PROFILE_METHOD_GET_Rx_HDCP_SUPPORT "getRxHDCPSupportedVersion"
+
+namespace WPEFramework
+{
+    namespace Plugin
+    {
+        SERVICE_REGISTRATION(LgiHdcpProfile, 1, 0);
+        static const char* getHdcpReasonStr (int eHDCPEnabledStatus);
+
+        LgiHdcpProfile::LgiHdcpProfile()
+        : HdcpProfile()
+        {
+            Register(HDCP_PROFILE_METHOD_GET_Rx_HDCP_SUPPORT, &LgiHdcpProfile::getRxHDCPSupportedVersion, this);
```

Comment:
Utils::Synchro::RegisterLockedApi should probably be used for method registration (see dt/helpers/UtilsSynchroIarm.hpp) - here and further on